### PR TITLE
Feature/create node

### PR DESCRIPTION
**Pull request by shivam-880**
### What changes were proposed in this pull request?
Add create node api to create node if doesn't already exits. Fail otherwise.
Removes the ability to create RemoteGraph directly, can now only be done through the client

### Why are the changes needed?
An alternative way to create node besides add node which updates the node if node already exists in the graph

### Does this PR introduce any user-facing change? If yes is this documented?
Documentation and Graphql docs

### How was this patch tested?
Unit tests

### Are there any further changes required?
No

### CODE REVIEW

File: python/tests/graphql/edit_graph/test_graphql.py [hunk 1]
```diff
@@ -382,6 +382,75 @@ def test_graph_properties_query():
...
+        query_nodes = """{graph(path: "g") {nodes {list {name}}}}"""
+        assert client.query(query_nodes) == {
+            "graph": {
+                "nodes": {
+                    "list": [{"name": "ben"}, {"name": "shivam"}]
+                }
+            }
+        }
+
+        remote_graph = RemoteGraph(path="g", client=client)
```
Comment (ljeub-pometry) [inline python/tests/graphql/edit_graph/test_graphql.py:438]: do we not have a `client.remote_graph(path="g")` method?

File: raphtory/src/db/graph/graph.rs [hunk 2]
```diff
@@ -3238,4 +3238,19 @@ mod db_tests {
         let graph = pool.install(|| Graph::new());
         assert_eq!(graph.core_graph().internal_num_nodes(), 0);
     }
+
+    #[test]
+    fn test_create_node() {
+        let g = Graph::new();
+        g.create_node(0, 1, [("test".to_string(), Prop::Bool(true))], None)
```
Comment (ljeub-pometry) [inline raphtory/src/db/graph/graph.rs:3245]: the `to_string` is not needed, right?

**Review by ljeub-pometry — APPROVED** [review]
please run tidy, otherwise looks good

**Review by miratepuffin — CHANGES_REQUESTED** [review]
Couple of small suggestions and then we can get it in :)

**Review by miratepuffin — APPROVED** [review]

**Pull request merged**